### ISSUE TITLE
`reportIncompatibleVariableOverride` does not follow PEP 591 in 1.1.145-1.1.147

### ISSUE THREAD

**Issue by ITProKyle** [microsoft/pyright#1964]
**Describe the bug**

Given the following code

```python
from typing import ClassVar

from typing_extensions import Final  # codebase needs to be compatible with python >= 3.7

class Base:
    EXAMPLE_CLASS_VAR: ClassVar[int] = 1

class ExampleFailure(Base):
    EXAMPLE_CLASS_VAR: Final[int] = 3
```

running pyright >= 1.1.145 results in `error: Instance variable "EXAMPLE_CLASS_VAR" overrides class variable of same name in class "Base" (reportIncompatibleVariableOverride)` target at the last line in the sample code snippet.

**Expected behavior**

pyright should infer that `ExampleFailure.EXAMPLE_CLASS_VAR` is a class variable since its value is defined in the body of the class. This is noted in the [Semantics and examples section of PEP 591](https://www.python.org/dev/peps/pep-0591/#semantics-and-examples).

> Type checkers should infer a final attribute that is initialized in a class body as being a class variable. Variables should not be annotated with both ClassVar and Final.

**Comment by erictraut** [comment]
Thanks for the bug report. I missed that statement in the PEP when I implemented support for Final. I've addressed the problem, and the fix will be in the next release.

**Comment by ITProKyle** [comment]
Thanks for the quick turnaround.

**Comment by erictraut** [comment]
This is included in pyright 1.1.148, which I just published. It will also be included in the next release of pylance.
